Issue-22: Fix issue with note text not getting exported in report.  B…

wonderificdave opened the pull request:
…oolean from the ternary expression was getting exported instead

asalant commented:
This one doesn't make sense to me. The change you made looks like the identical logic to the ternary expression you replaced - "If the note is nil, append nil. Otherwise append note.text."

What was the issue you were trying to fix? Can you reproduce it with a test that fails with the old code and passes with the new?

wonderificdave commented:
Hi Alon, can you take a look at this one again?  I added a test and also simplified the fix... it is an order of operations problem.  The new fix is similar to the code in visit.rb line 64

wonderificdave commented:
Hi Alon,

Can you take a look at this one again?  I put a fix in a few weeks ago and
wanted to make sure you saw it.  Let me know if you have any questions
-dave

On Sun, May 1, 2016 at 10:17 PM Alon Salant <notifications@github.com>
wrote:

> This one doesn't make sense to me. The change you made looks like the
> identical logic to the ternary expression you replaced - "If the note is
> nil, append nil. Otherwise append note.text."
>
> What was the issue you were trying to fix? Can you reproduce it with a
> test that fails with the old code and passes with the new?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/asalant/freehub/pull/26#issuecomment-216106568>
>

asalant commented:
Thanks for the poke. I started to check it out on a new laptop and got hung
up on @$!#@! Rails/Yosemite setup. I'll look again this week. Don't
hesitate to poke me again.

On Mon, May 23, 2016 at 1:53 PM, wonderificdave <notifications@github.com>
wrote:

> Hi Alon,
>
> Can you take a look at this one again? I put a fix in a few weeks ago and
> wanted to make sure you saw it. Let me know if you have any questions
> -dave
>
> On Sun, May 1, 2016 at 10:17 PM Alon Salant <notifications@github.com>
> wrote:
>
> > This one doesn't make sense to me. The change you made looks like the
> > identical logic to the ternary expression you replaced - "If the note is
> > nil, append nil. Otherwise append note.text."
> >
> > What was the issue you were trying to fix? Can you reproduce it with a
> > test that fails with the old code and passes with the new?
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly or view it on GitHub
> > <https://github.com/asalant/freehub/pull/26#issuecomment-216106568>
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/asalant/freehub/pull/26#issuecomment-221092835>
>

asalant commented on app/models/service.rb:
```diff
@@ -74,7 +74,7 @@ def to_csv
     values << volunteered
     values << paid
-    values << note.nil? ? nil : note.text
+    values << (note.nil? ? nil : note.text)
```
Hah! Great fix. Esp with the test.

asalant commented:
Resolved in PR #28.

wonderificdave commented:
Hi Alon,

Looks like the freehub site is down with a 500 error.  Have you seen this?
Let me know if there's anything I can do to help

Thanks!
-dave

On Mon, May 23, 2016 at 9:49 PM Alon Salant <notifications@github.com>
wrote:

> Closed #26 <https://github.com/asalant/freehub/pull/26>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/asalant/freehub/pull/26#event-669798905>
>

asalant commented:
Hey Dave, thanks for the heads up and the offer to help. I'm on it but
would be great to be get in touch directly. Please send me your email
address to alon at salant.org. Thanks!

On Tue, May 24, 2016 at 9:44 PM, wonderificdave <notifications@github.com>
wrote:

> Hi Alon,
>
> Looks like the freehub site is down with a 500 error. Have you seen this?
> Let me know if there's anything I can do to help
>
> Thanks!
> -dave
>
> On Mon, May 23, 2016 at 9:49 PM Alon Salant <notifications@github.com>
> wrote:
>
> > Closed #26 <https://github.com/asalant/freehub/pull/26>.
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly or view it on GitHub
> > <https://github.com/asalant/freehub/pull/26#event-669798905>
> >
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/asalant/freehub/pull/26#issuecomment-221471834>
>
